src/ImageSearchCriteria.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+
+namespace App;
+
+
+use App\Exceptions\InvalidLimitException;
+use App\Exceptions\InvalidPageException;
+use App\Exceptions\NonexistentOrderByColumn;
+use App\Exceptions\NonexistentOrderingType;
+
+class ImageSearchCriteria
+{
+    private $tag;
+    private $page;
+    private $limit;
+    private $order;
+    private $ascDesc;
+
+    /**
+     * ImageSearchCriteria constructor.
+     * @param $tag
+     * @param $page
+     * @param $limit
+     * @param $order
+     * @param $ascDesc
+     */
+    public function __construct($tag, $page, $limit, $order, $ascDesc)
+    {
+        $this->tag = $tag;
+        if ($page <= 0) {
+            throw new InvalidPageException("Page must be positive");
+        } else {
+            $this->page = $page;
+        }
+        if ($limit <= 0) {
+            throw new InvalidLimitException("Limit must be positive");
+        } else {
+            $this->limit = $limit;
+        }
+        if ($order !== 'id') {
+            throw new NonexistentOrderByColumn("Nonexistent column name");
+        } else {
+            $this->order = $order;
+        }
+        if ($ascDesc !== 'ASC' and $ascDesc !== 'DESC') {
```

Comment:
`and`

Reply:
Every "and" replaced with &&. "Or" nor found